### PR TITLE
Add SlideshowProvider

### PR DESCRIPTION
**Pull request by michaelwehrley**
* No longer have `LessonModule` check first provider to determine if it has a `SlideshowProvider`
* Have `LessonModule` check for a the exist of a `SlideshowProvider` via `find_by`
* Add spec to check for creation of `SlideshowProvider`

[Finishes: #114693699]

### CODE REVIEW

File: app/models/content_modules/lesson_module.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module ContentModules
   # Container for didactic content.
   class LessonModule < BitCore::ContentModule
+    CONTENT_PROVIDER_TYPE = "BitCore::ContentProviders::SlideshowProvider"
```
Comment (ericcf) [inline app/models/content_modules/lesson_module.rb]: Would it be clearer to name this `SLIDESHOW_PROVIDER_TYPE` or similar?

File: spec/models/content_modules/lesson_module_spec.rb [hunk 2]
```diff
@@ -34,5 +34,29 @@ module ContentModules
         end
       end
     end
+
+    describe ".build_slide" do
+      describe "when the first content provider isn't of type `SlideshowProvider`" do
+        it "will add a slideshow content provider" do
+          BitCore::ContentProvider.create!(
```
Comment (ericcf) [inline spec/models/content_modules/lesson_module_spec.rb]: Any reason not to `ContentProviders::LearnLessonsIndexProvider.create!(...` instead?

Comment (michaelwehrley) [inline spec/models/content_modules/lesson_module_spec.rb]: nope, seems helpful

File: app/models/content_modules/lesson_module.rb (hunk 1 above)
Comment (michaelwehrley) [inline app/models/content_modules/lesson_module.rb]: i like. thanks.

**Comment by ericcf** [comment]
:shipit:

**Pull request merged**